zappa/core.py

```diff
@@ -2919,7 +2919,15 @@ def get_scheduled_event_name(self, event, function, lambda_name, index=0):
         """
         Returns an AWS-valid CloudWatch rule name using a digest of the event name, lambda name, and function.
         This allows support for rule names that may be longer than the 64 char limit.
+
+        function pattern: '^[._A-Za-z0-9]{,63}$'
         """
+        function_regex = re.compile("^[._A-Za-z0-9]{,63}$")
```

Comment:
You should need the "\." here to allow a "." (period), as "." alone matches all characters except new lines.

Reply:
Also, I'm unclear on what `{,63}` does.  I assume that it allows any number of characters up to 63, is that correct?

If that's the case it may be clearer to write `{0,63}`.

Reply:
> You should need the "." here to allow a "." (period), as "." alone matches all characters except new lines.

I got the following warning message and removed the escape
> Redundant character escape '\.' in RegExp 

> If that's the case it may be clearer to write {0,63}.
ok!